src/Sylius/Bundle/UserBundle/Form/EventSubscriber/SetCustomerFormSubscriber.php

```diff
@@ -0,0 +1,64 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Paweł Jędrzejewski
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Sylius\Bundle\UserBundle\Form\EventSubscriber;
+
+use Sylius\Component\Resource\Repository\RepositoryInterface;
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+use Symfony\Component\Form\FormEvent;
+use Symfony\Component\Form\FormEvents;
+
+/**
+ * @author Anna Walasek <anna.walasek@lakion.com>
+ */
+class SetCustomerFormSubscriber implements EventSubscriberInterface
+{
+    /**
+     * @var RepositoryInterface
+     */
+    private $customerRepository;
+
+    /**
+     * @param RepositoryInterface $customerRepository
+     */
+    public function __construct(RepositoryInterface $customerRepository)
+    {
+        $this->customerRepository = $customerRepository;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public static function getSubscribedEvents()
+    {
+        return [ 
+            FormEvents::PRE_SUBMIT => 'preSubmit',
+        ];
+    }
+
+    public function preSubmit(FormEvent $event) 
+    {
+        $data = $event->getData();
+
+        if (null === $data['email']) {
```

Comment:
Is it possible that this variable will be not set? It will be passed as empty string or string with email.